### PR TITLE
Fix coverage

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Test PR follow up https://github.com/tonerdo/coverlet/issues/700

This is a sample, it's not intended for merge.

I did some debugging and maybe found the issue(we'll see on this PR)

@antonfirsov PR add some out of process tests that slow down a bit the shutdown of test process, you're using msbuild that is affected by know issue https://github.com/tonerdo/coverlet/blob/master/Documentation/KnowIssues.md#1-vstest-stops-process-execution-earlydotnet-test

So new out of process tests create/update hits file(file where we put line/branch hit) on process exits but all other tests fail to log hits because vs test kill test process(the parent process of all other out of ones) before flush https://github.com/microsoft/vstest/issues/1900#issuecomment-457488472

This PR change the drivers to collectors

BTW vstest has got a bug that I've fixed in https://github.com/microsoft/vstest/pull/2221 but not yet released so you need for now to use the preview test sdk.

My local result
image

cc: @JimBobSquarePants

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
@JimBobSquarePants got and error on codecov upload
```
2020-01-25T14:11:49.1313263Z ==> Scanning for reports
2020-01-25T14:11:49.5779598Z     + d:\a\ImageSharp\ImageSharp\tests\ImageSharp.Tests\TestResults\6d6166d5-6a42-46eb-93c3-6f4b736c89aa\coverage.info
2020-01-25T14:11:49.5787040Z     + d:\a\ImageSharp\ImageSharp\artifacts\bin\tests\ImageSharp.Tests.ProfilingSandbox\Release\netcoreapp3.1\win7-x64\Microsoft.VisualStudio.CodeCoverage.Shim.dll
2020-01-25T14:11:49.5793378Z     + d:\a\ImageSharp\ImageSharp\artifacts\bin\tests\ImageSharp.Tests\Release\netcoreapp3.1\Microsoft.VisualStudio.CodeCoverage.Shim.dll
2020-01-25T14:11:49.5793911Z ==> Uploading reports
2020-01-25T14:11:49.7850673Z     HTTP 400
2020-01-25T14:11:49.7851344Z Please provide the repository token to upload reports via `-t :repository-token`
```

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -85,11 +85,14 @@ jobs:
...
+            # Avoid "Please provide the repository token to upload reports via `-t :repository-token`"
+            # https://community.codecov.io/t/whitelist-github-action-servers-to-upload-without-a-token/491/10
+            # https://github.community/t5/GitHub-Actions/Make-secrets-available-to-builds-of-forks/m-p/42814/highlight/true#M5129
             - name: Update Codecov
               uses: iansu/codecov-action-node@v1.0.0
-              if: matrix.options.codecov == true
+              if: matrix.options.codecov == true && startsWith(github.repository, 'SixLabors')
               with:
-                  token: ${{secrets.CODECOV_TOKEN}}
+                  token: 0ef021c7-2679-4012-b42f-4bed33d99450
```
Comment (JimBobSquarePants) [inline .github/workflows/build-and-test.yml]: I'm not concerned about exposing this temporarily.

**Comment by JimBobSquarePants** [comment]
@MarcoRossignoli That's fantastic thank you! 

I can't see any reason why we cannot merge this directly. Is there more you intend to add?

**Comment by MarcoRossignoli** [comment]
> I can't see any reason why we cannot merge this directly. Is there more you intend to add?

Didn't know if was your workflow idea. BTW  good!

This issue is the reason for us to move most user as possible to collectors, this bug is hard to discover and is related to very "small" behaviour changes.

File: tests/Directory.Build.targets [hunk 2]
```diff
@@ -23,30 +23,14 @@
...
     <!--Test Dependencies-->
     <PackageReference Update="BenchmarkDotNet" Version="0.12.0" />
     <PackageReference Update="Colourful" Version="2.0.3" />
-    <PackageReference Update="coverlet.collector" Version="1.1.0" PrivateAssets="All"/>
-    <PackageReference Update="coverlet.msbuild" Version="2.8.0" PrivateAssets="All"/>
+    <PackageReference Update="coverlet.collector" Version="1.2.0" PrivateAssets="All"/>
     <PackageReference Update="Magick.NET-Q16-AnyCPU" Version="7.14.4" />
     <PackageReference Update="Microsoft.DotNet.RemoteExecutor" Version="5.0.0-beta.20069.1" />
-    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="16.4.0" />
+    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="16.5.0-preview-20200116-01" />
```
Comment (MarcoRossignoli) [inline tests/Directory.Build.targets]: @JimBobSquarePants I suggest to add a comment above this line and update with "stable" release in future.

Comment (JimBobSquarePants) [reply tests/Directory.Build.targets]: Ah right, yeah. Good idea.

**Pull request merged**